### ISSUE TITLE
Failed to spawn: unexpectedly timed out while waiting for signal from process

### ISSUE THREAD

**Issue by LaiKash** [frida/frida#2516]
As the tittle says, with latest Frida version (16.0.19 both for tools and in Android):

![image](https://github.com/frida/frida/assets/24409595/be0737b2-ac35-42d5-b1bc-c3acb652ad43)

In this version it is giving a "timed out" error, in slightly previous ones it was not the case (just stays in a loop).

My device specs (Android 13, could be related to other issues), currently running the latest Xiaomi.eu ROM:

![image](https://github.com/frida/frida/assets/24409595/b1b6b0a7-c9e2-4764-a89d-b4e8be401e5a)

There is nothing relevant in Logcat and I have already tried previous Frida (both tools and core) versions. None of them worked, core is not crashing but sometimes I get a soft reboot. Please, let me know if something else is needed.

**Comment by fuqiuluo** [comment]
I have the same problem,  
![image](https://github.com/frida/frida/assets/61898844/fd44e2bf-93d5-4dc4-8abd-03c39f857acd)

**Comment by Rongrkang** [comment]
I have the same problem,
![image](https://github.com/frida/frida/assets/8871800/552de118-eabe-49d3-a56b-c01b50a1d7f3)

**Comment by 4xi0ng** [comment]
same issue, after disable magiskHide and Zygisk, it works!

**Comment by LaiKash** [comment]
> same issue, after disable magiskHide and Zygisk, it works!

Not for me, at least. Both with Zygisk and magiskHide disabled, not working

**Comment by tconn87** [comment]
> I tried this [frida/frida-core#376 (comment)](https://github.com/frida/frida-core/issues/376#issuecomment-896764075) Set the value to false and checked for running processes of "usap" and killed them all. It worked for me after that. I may have also killed and restarted the frida-server for good measure

maybe don't do this. My power button doesn't let me reboot, none of my apps will load, Accessibility Menu doesn't open, Power+VolDwn doesn't reboot, and I lost my adb connection

**Comment by mriamnobody** [comment]
> @LaiKash @Rongrkang @fuqiuluo You can try this:
> 
> 1. getprop|grep usap
> 2. If getprop xxx.usap returns true
> 3. setprop xxxx.usap false
> 4. And then getprop xxxx.usap will return false
> 5. Use spawn by your own way

In case if the workaround provided by @aabbccgg somehow doesn't work, there are a few steps you can follow to check if it helps. Start `adb shell` in root and run the following command `ps -A | grep usap` and check if the `usap` is running. If it is running then run the command `pkill -f usap`, this will kill all the instances of the `usap`.

**Comment by mriamnobody** [comment]
> > @LaiKash @Rongrkang @fuqiuluo You can try this:
> > 
> > 1. getprop|grep usap
> > 2. If getprop xxx.usap returns true
> > 3. setprop xxxx.usap false
> > 4. And then getprop xxxx.usap will return false
> > 5. Use spawn by your own way
> 
> In case if the workaround provided by @aabbccgg somehow doesn't work, there are a few steps you can follow to check if it helps. Start `adb shell` in root and run the following command `ps -A | grep usap` and check if the `usap` is running. If it is running then run the command `pkill -f usap`, this will kill all the instances of the `usap`.

As a better way to resolve the issue will be to first disable the `usap` (before starting the frida server) as using the command mentioned here https://github.com/frida/frida-core/issues/376#issuecomment-896764075

**Comment by mriamnobody** [comment]
> > > @LaiKash @Rongrkang @fuqiuluo You can try this:
> > > 
> > > 1. getprop|grep usap
> > > 2. If getprop xxx.usap returns true
> > > 3. setprop xxxx.usap false
> > > 4. And then getprop xxxx.usap will return false
> > > 5. Use spawn by your own way
> > 
> > 
> > In case if the workaround provided by @aabbccgg somehow doesn't work, there are a few steps you can follow to check if it helps. Start `adb shell` in root and run the following command `ps -A | grep usap` and check if the `usap` is running. If it is running then run the command `pkill -f usap`, this will kill all the instances of the `usap`.
> 
> As a better way to resolve the issue will be to first disable the `usap` (before starting the frida server) as using the command mentioned here [frida/frida-core#376 (comment)](https://github.com/frida/frida-core/issues/376#issuecomment-896764075)

This is to avoid the issues as stated by @tconn87 https://github.com/frida/frida/issues/2516#issuecomment-1641450009

**Comment by mriamnobody** [comment]
So to summarize first reboot the device then disable the `usap` using the command provided at https://github.com/frida/frida-core/issues/376#issuecomment-896764075, after disabling `usap` start your frida server and it should work. If even after performing the above-mentioned steps if Frida is still not working then you should take a look at https://github.com/frida/frida/issues/1072 and https://github.com/frida/frida/issues/662. It basically says to disable Magisk hide.

**Comment by aabbccgg** [comment]
> > > @LaiKash @Rongrkang @fuqiuluo You can try this:
> > > 
> > > 1. getprop|grep usap
> > > 2. If getprop xxx.usap returns true
> > > 3. setprop xxxx.usap false
> > > 4. And then getprop xxxx.usap will return false
> > > 5. Use spawn by your own way
> > 
> > 
> > In case if the workaround provided by @aabbccgg somehow doesn't work, there are a few steps you can follow to check if it helps. Start `adb shell` in root and run the following command `ps -A | grep usap` and check if the `usap` is running. If it is running then run the command `pkill -f usap`, this will kill all the instances of the `usap`.
> 
> As a better way to resolve the issue will be to first disable the `usap` (before starting the frida server) as using the command mentioned here [frida/frida-core#376 (comment)](https://github.com/frida/frida-core/issues/376#issuecomment-896764075)

OEMs may have modified the property name of usap. So we should use these command following:
getprop|grep usap
setprop xxxx.usap false

The command here [https://github.com/frida/frida-core/issues/376#issuecomment-896764075](url) may not work well in some OEMs' OS.

**Comment by jmpangilinan** [comment]
Got the same Issue, It works on -U with frida-gadget.so on the python folder, but when I try remotely without USB debugging, the app won't launch, it works if I execute the spawn command with -n "AppName" then simultaneously opening the app on my device, but sometimes it work and sometimes not and just say "crash failed to allocate memory"

**Comment by iamvinny** [comment]
It works when I disable Zygisk, the problem is that I need Zygisk...

**Comment by aabbccgg** [comment]
> > @LaiKash @Rongrkang @fuqiuluo You can try this:
> > 
> > 1. getprop|grep usap
> > 2. If getprop xxx.usap returns true
> > 3. setprop xxxx.usap false
> > 4. And then getprop xxxx.usap will return false
> > 5. Use spawn by your own way
> 
> ```
> olives:/ # getprop|grep usap
> [persist.device_config.runtime_native.usap_pool_enabled]: [false]
> [persist.device_config.runtime_native.usap_pool_refill_delay_ms]: [3000]
> [persist.device_config.runtime_native.usap_pool_size_max]: [10]
> [persist.device_config.runtime_native.usap_pool_size_min]: [1]
> [persist.device_config.runtime_native.usap_refill_threshold]: [5]
> olives:/ #
> ```
> 
> but failed, im using android 10 .

Yes there are also have other problems can lead to the spawn crash error. Can you post an error log?

**Comment by LaiKash** [comment]
> Got the same Issue, It works on -U with frida-gadget.so on the python folder, but when I try remotely without USB debugging, the app won't launch, it works if I execute the spawn command with `-n <AppName>` then simultaneously opening the app on my device, but sometimes it work and sometimes not and just say "crash failed to allocate memory", it happens when the app opens first then the attaching is late, needs to be simultaneously.

Same here, the other solutions didn't work (disabling Zygisk, usap and magisk hide) @mriamnobody @aabbccgg 

It seems to be an error with addresses. When running Frida server, I get the following error message:

```{"type":"error","description":"Error: invalid address","stack":"Error: invalid address\n    at Object.value [as patchCode] (frida/runtime/core.js:207:1)\n    at ln (frida/node_modules/frida-java-bridge/lib/android.js:1209:1)\n    at pn.activate (frida/node_modules/frida-java-bridge/lib/android.js:1275:1)\n    at mn.replace (frida/node_modules/frida-java-bridge/lib/android.js:1323:1)\n    at Function.set [as implementation] (frida/node_modules/frida-java-bridge/lib/class-factory.js:1185:1)\n    at Function.set [as implementation] (frida/node_modules/frida-java-bridge/lib/class-factory.js:1099:1)\n    at installLaunchTimeoutRemovalInstrumentation (/internal-agent.js:435:37)\n    at init (/internal-agent.js:51:3)\n    at c.perform (frida/node_modules/frida-java-bridge/lib/vm.js:12:1)\n    at _performPendingVmOps (frida/node_modules/frida-java-bridge/index.js:250:1)","fileName":"frida/runtime/core.js","lineNumber":207,"columnNumber":1}```

**Comment by durori** [comment]
This is what i do for samsung A50 and worked

```
adb shell
su
mount -o remount,rw '/'
chmod 777 /
adb shell "su -c setenforce 0"
setprop persist.device_config.runtime_native.usap_pool_enabled false
settings put system screen_off_timeout 100000000
exit
exit
```

now you can run frida-server
terminal 1 (use cmd for windows)
`adb shell "su -c /data/local/tmp/frida-serverarm64"`
terminal 2 (maybe powershell for windows)
`frida -U -f package.app -l script.js`

Good Luck

**Comment by SafaSafari** [comment]
https://github.com/frida/frida/issues/2719#issuecomment-1869274133

**Comment by gauravssnl** [comment]
I installed the frida-server-16.1.11-android-arm64 version first , it worked for a while and I tried to run the below command as per Frida doc : frida-trace -U -i open com.android.chrome , then my phone crashed and restarted.

**What work for me**

I have installed the frida-server-16.1.11-linux-arm64 ( **Linux version** ) and that works for me and phone is not crashing. You all can give this solution a try and let me know.

Note : my phone architecture is ARM64, please install the corresponding Linux version according to your phone architecture.

Let me know your results.

**Comment by shunf4** [comment]
From https://www.coolapk.com/feed/56655501:

> #一加12# #frida#
> 记个小坑，使用frida spawn模式时会time out，提示PID 1442进程超时，使用frida-ps查询发现，该进程是zygote，遂想到一加12不支持32位应用，于是使用adb shell su -c stop zygote_secondary结束该进程，再次spawn，运行成功。

Translated:

#OnePlus 12# #frida#
Note a small pitfall. When using frida spawn mode, it will time out, prompting PID 1442 process timeout. Using frida-ps to query, it is found that the process is zygote. So I thought that OnePlus 12 does not support 32-bit applications, so I used adb shell su -c stop zygote_secondary to end the process, and spawned again, and it ran successfully.

TL;DR: Kill/Stop other zygote(-like) processes/services that is not zygote64.

**Comment by crifan** [comment]
Similar issue:
* Environment
  * PC: Mac M2 Max, Frida 16.5.6
  * Mobile:Xiaomi 8, LineageOS 19, Android 12, Frida 16.5.6 (MagiskFrida）
* Error
```bash
frida -U -f com.xxx.yyy -l hook_zzz.js

   . . . .   Connected to MI 8 (id=f97a0408)
Failed to spawn: unexpectedly timed out while waiting for signal from process with PID 5168
```
* Solution
  * Disable: `Zygisk` (in Kitsune Mask)

![8FBA80EA-9926-43AA-B25B-BC2692A2F01D](https://github.com/user-attachments/assets/a12a631d-3ab8-403b-ae35-5d61d8a737c1)

![9694663F-9D66-43DB-A36C-9F4F1198B7FE](https://github.com/user-attachments/assets/1a8b5802-f43d-4236-b9ba-706a5632b0b7)

**Comment by Icyrockton** [comment]
For users with OnePlus(一加) running ColorOS 15, please use the following command to terminate the zygote32 process. Otherwise, Frida will wait for the signal response from the zygote32 process (obviously timed out) instead of the expected zygote64 process.

```
setprop persist.sys.tango_zygote32.start 0
```

After this, `zygote_tango` will be stopped.

```
OP5CFBL1:/data/local/tmp # getprop | grep "zygote"
[init.svc.zygote]: [running]
[init.svc.zygote_ocomp]: [running]
[init.svc.zygote_tango]: [stopped]
[init.svc_debug_pid.zygote]: [26502]
[init.svc_debug_pid.zygote_ocomp]: [1611]
[init.svc_debug_pid.zygote_tango]: []
[persist.oplus.zygote_ocomp]: [true]
[persist.sys.system.abnormalboot]: [2025-01-15 07:44:49:zygote reboot by netd]
[persist.sys.tango_zygote32.start]: [0]
[ro.boottime.zygote]: [6197081507]
[ro.boottime.zygote_ocomp]: [6380248329]
[ro.boottime.zygote_tango]: [6382967496]
[ro.zygote]: [zygote64]
```

**Comment by dailybaldman** [comment]
一加Ace5 也是这种情况, 我安装了magisk, 开启了zygisk, 也是无法spawn任意app, 报超时错误, 但是attach可以.

**Comment by dailybaldman** [comment]
> For users with OnePlus(一加) running ColorOS 15, please use the following command to terminate the zygote32 process. Otherwise, Frida will wait for the signal response from the zygote32 process (obviously timed out) instead of the expected zygote64 process.
> 
> ```
> setprop persist.sys.tango_zygote32.start 0
> ```
> 
> After this, `zygote_tango` will be stopped.
> 
> ```
> OP5CFBL1:/data/local/tmp # getprop | grep "zygote"
> [init.svc.zygote]: [running]
> [init.svc.zygote_ocomp]: [running]
> [init.svc.zygote_tango]: [stopped]
> [init.svc_debug_pid.zygote]: [26502]
> [init.svc_debug_pid.zygote_ocomp]: [1611]
> [init.svc_debug_pid.zygote_tango]: []
> [persist.oplus.zygote_ocomp]: [true]
> [persist.sys.system.abnormalboot]: [2025-01-15 07:44:49:zygote reboot by netd]
> [persist.sys.tango_zygote32.start]: [0]
> [ro.boottime.zygote]: [6197081507]
> [ro.boottime.zygote_ocomp]: [6380248329]
> [ro.boottime.zygote_tango]: [6382967496]
> [ro.zygote]: [zygote64]
> ```

thanks!! it works!!!!!

**Comment by anasfanani** [comment]
```
anasfanani ➜ ~/workspaces $ frida -U -f com.android.vending
     ____
    / _  |   Frida 16.7.11 - A world-class dynamic instrumentation toolkit
   | (_| |
    > _  |   Commands:
   /_/ |_|       help      -> Displays the help system
   . . . .       object?   -> Display information about 'object'
   . . . .       exit/quit -> Exit
   . . . .
   . . . .   More info at https://frida.re/docs/home/
   . . . .
   . . . .   Connected to 24069PC21G (id=192.168.1.10:5555)
Failed to spawn: unexpectedly timed out while waiting for signal from process with PID 1477
```

Frida try spawn with PID 1477 which is tango_translator.
Tango is an ARM binary translator used on GNU/Linux and Android. It enables 32-bit ARM programs to run on 64-bit only ARM processors. 

```
anasfanani ➜ ~/workspaces $ adb shell 
peridot:/ $ su
peridot:/ # ps -o PID,NAME,COMMAND,CMDLINE -Aww | grep zygote | grep -v grep
 1477 zygote                      /system_ext/bin/tango_translator zygote
16870 zygote64                    /system/bin/app_process64   zygote64
17844 webview_zygote              /system/bin/app_process64   webview_zygote
peridot:/ # 
```

Its maybe from this https://github.com/frida/frida-core/blob/0d6c0c4e60ab633b2595eb8c148f3b91ca9e78b0/lib/payload/fork-monitor.vala#L63 
Frida try to find with this order : zygote, zygote64, usap32, usap64.

To run frida with desired zygote process we can kill process which fail to spawn.

> From https://www.coolapk.com/feed/56655501:
> 
> > #一加12# #frida#
> > 记个小坑，使用frida spawn模式时会time out，提示PID 1442进程超时，使用frida-ps查询发现，该进程是zygote，遂想到一加12不支持32位应用，于是使用adb shell su -c stop zygote_secondary结束该进程，再次spawn，运行成功。
> 
> Translated:
> 
> #OnePlus 12# #frida# Note a small pitfall. When using frida spawn mode, it will time out, prompting PID 1442 process timeout. Using frida-ps to query, it is found that the process is zygote. So I thought that OnePlus 12 does not support 32-bit applications, so I used adb shell su -c stop zygote_secondary to end the process, and spawned again, and it ran successfully.
> 
> TL;DR: Kill/Stop other zygote(-like) processes/services that is not zygote64.

Yes but some app will randomly force close, instead kill the process I do bind mount the pid, restart frida-server, try spawn, after first spawn success, I do restore the process because if first spawn success, next spawn also success no matter zygote pid is back, I don't know why, this is my example snippet to run frida-server.

```sh
tmpfile=$(mktemp -d)
mount -o bind $tmpfile /proc/1477
pkill -9 -f frida-server
/data/local/tmp/frida-server -l 0.0.0.0:27042 -D
```

Then try spawn

<details>
  <summary>See this command log.</summary>

```
anasfanani ➜ ~/workspaces $ adb shell
peridot:/ $ su                                                                  
peridot:/ # ps -o PID,NAME,COMMAND,CMDLINE -Aww | grep zygote | grep -v grep
 1477 zygote                      /system_ext/bin/tango_translator zygote
16870 zygote64                    /system/bin/app_process64   zygote64
17844 webview_zygote              /system/bin/app_process64   webview_zygote
peridot:/ #
peridot:/ # tmpfile=$(mktemp -d)
peridot:/ # echo $tmpfile
/data/local/tmp/tmp.IeU5qi6rFI
peridot:/ # mount -o bind $tmpfile /proc/1477
peridot:/ #
peridot:/ # ps -o PID,NAME,COMMAND,CMDLINE -Aww | grep zygote | grep -v grep
16870 zygote64                    /system/bin/app_process64   zygote64
17844 webview_zygote              /system/bin/app_process64   webview_zygote
peridot:/ #
peridot:/ # pkill -9 -f frida-server
1|peridot:/ # /data/local/tmp/frida-server -l 0.0.0.0:27042 -D
peridot:/ # exit
peridot:/ $ exit
anasfanani ➜ ~/workspaces $ frida -U -f com.android.vending
     ____
    / _  |   Frida 16.7.11 - A world-class dynamic instrumentation toolkit
   | (_| |
    > _  |   Commands:
   /_/ |_|       help      -> Displays the help system
   . . . .       object?   -> Display information about 'object'
   . . . .       exit/quit -> Exit
   . . . .
   . . . .   More info at https://frida.re/docs/home/
   . . . .
   . . . .   Connected to 24069PC21G (id=192.168.1.10:5555)
Spawned `com.android.vending`. Resuming main thread!                    
[24069PC21G::com.android.vending ]-> exit

Thank you for using Frida!
anasfanani ➜ ~/workspaces $ adb shell
peridot:/ $ su
peridot:/ # umount /proc/1477
peridot:/ # ps -o PID,NAME,COMMAND,CMDLINE -Aww | grep zygote | grep -v grep
 1477 zygote                      /system_ext/bin/tango_translator zygote
16870 zygote64                    /system/bin/app_process64   zygote64
17844 webview_zygote              /system/bin/app_process64   webview_zygote
peridot:/ # exit
peridot:/ $ exit
anasfanani ➜ ~/workspaces $ frida -U -f com.termux
     ____
    / _  |   Frida 16.7.11 - A world-class dynamic instrumentation toolkit
   | (_| |
    > _  |   Commands:
   /_/ |_|       help      -> Displays the help system
   . . . .       object?   -> Display information about 'object'
   . . . .       exit/quit -> Exit
   . . . .
   . . . .   More info at https://frida.re/docs/home/
   . . . .
   . . . .   Connected to 24069PC21G (id=192.168.1.10:5555)
Spawned `com.termux`. Resuming main thread!                             
[24069PC21G::com.termux ]-> exit

Thank you for using Frida!
anasfanani ➜ ~/workspaces $ 
```
</details>

Another way to mitigate this issue is just hide the zygote pid using hacky way, maybe using `nsenter`, `libprocesshider.so`, `zygisk` module to hide pid, the best way is the dev fix this or someone make PR, frida-core should make another option if unexpectedly timed out happen.



<details>
  <summary>I've tried 50 frida-server version, you may want to try it.</summary>

```sh
#!/bin/sh

API_RELEASE_URL="https://api.github.com/repos/frida/frida/releases?per_page=50"
RELEASES=$(curl -s $API_RELEASE_URL)
# RELEASES=$(cat release_cache.json)
OS="android"
ARCH="arm64"
VARIANT="frida-server"
DOWNLOADS_DIR="downloads"
mkdir -p "$DOWNLOADS_DIR"

rm -f "${0}.log"
mkfifo "$HOME/.pipe.$$"
tee -a "${0}.log" <"$HOME/.pipe.$$" &
exec 1>"$HOME/.pipe.$$" 2>&1
rm -f "$HOME/.pipe.$$"

run_frida_server(){
    echo
    printf "Run frida-server $tag to android\n"
    adb wait-for-device
    while [ "$(adb get-state)" = "offline" ]; do
        echo "Device is offline. Waiting..."
        sleep 1
    done
    adb shell su -c "pkill -9 -f frida-server"
    adb shell su -c "rm -f /data/local/tmp/frida-server"
    adb push "$DOWNLOADS_DIR/$1" /data/local/tmp/frida-server
    adb shell su -c "chmod +x /data/local/tmp/frida-server"
    set -x
    adb shell su -c "/data/local/tmp/frida-server -l 0.0.0.0:27042 -v > /data/local/tmp/${1}.log 2>&1 &"
    frida -U -f com.termux
    adb shell su -c cat "/data/local/tmp/${1}.log"
    set +x
}

for tag in $(echo "$RELEASES" | jq -r '.[].tag_name'); do
    DOWNLOAD_URL=$(
        echo "$RELEASES" \
        | jq -r --arg tag "$tag" \
        --arg os "$OS" \
        --arg arch "$ARCH" \
        --arg variant "$VARIANT" '
        .[] | select(.tag_name == $tag) | 
        .assets[] | select(.name | contains($variant) and contains($os) and contains($arch)) | 
        .browser_download_url
        '
    )
    archive_name=$(basename "$DOWNLOAD_URL")
    bin_name=$(basename "$archive_name" .xz)
    if [ ! -f "$DOWNLOADS_DIR/$bin_name" ]; then
        printf "Downloading from: $DOWNLOAD_URL\n"
        curl -LO "$DOWNLOAD_URL"
        unxz -f "$archive_name"
        mv "$bin_name" "$DOWNLOADS_DIR"
    fi
    run_frida_server "$bin_name"
done
```
</details>

**Comment by n08i40k** [comment]
> For users with OnePlus(一加) running ColorOS 15, please use the following command to terminate the zygote32 process. Otherwise, Frida will wait for the signal response from the zygote32 process (obviously timed out) instead of the expected zygote64 process.

Thank you very much. I tried to disable zygote32 using Magisk module (Xiaomi on custom A15 rom) and everything started working as it should.

post-fs-data.sh content:
```
#!/system/bin/sh     

resetprop ro.zygote zygote64

resetprop ro.odm.product.cpu.abilist arm64-v8a
resetprop ro.odm.product.cpu.abilist64 arm64-v8a
resetprop ro.product.cpu.abi arm64-v8a
resetprop ro.product.cpu.abilist arm64-v8a
resetprop ro.product.cpu.abilist64 arm64-v8a
resetprop ro.system.product.cpu.abilist arm64-v8a
resetprop ro.system.product.cpu.abilist64 arm64-v8a
resetprop ro.vendor.product.cpu.abilist arm64-v8a
resetprop ro.vendor.product.cpu.abilist64 arm64-v8a

resetprop -d ro.odm.product.cpu.abilist32
resetprop -d ro.product.cpu.abilist32
resetprop -d ro.system.product.cpu.abilist32
resetprop -d ro.vendor.product.cpu.abilist32

# Without this there will be a bootloop
for BINARY in \
  /system/bin/app_process32 \
  /system/bin/zygote32 \
  /system/xbin/app_process32 \
  /system/xbin/zygote32; do
    if [ -e "$BINARY" ]; then
      mount -o bind /dev/null "$BINARY"
    fi
done
```

**Comment by suryasarisa99** [comment]
> same issue, after disable magiskHide and Zygisk, it works!

its works for me after disable zygisk and by settings this: `setprop persist.device_config.runtime_native.usap_pool_enabled false`
but the problem if i turn off zygisk the app detects root, so i can't even open it. how to handle this situation ?

**Comment by suryasarisa99** [comment]
> It works when I disable Zygisk, the problem is that I need Zygisk...

yes same problem, if i turn of zygisk root the app detects root. so i want to hide root and use frida at same time. do u find any solution ?

**Comment by HaclyonDev** [comment]
> > It works when I disable Zygisk, the problem is that I need Zygisk...
> 
> yes same problem, if i turn of zygisk root the app detects root. so i want to hide root and use frida at same time. do u find any solution ?

> You can try other versions of Frida. My phone is Android 12, and after switching to version 16.1.4, it works normally

> > It works when I disable Zygisk, the problem is that I need Zygisk...
> 
> yes same problem, if i turn of zygisk root the app detects root. so i want to hide root and use frida at same time. do u find any solution ?

You can try other versions of Frida. My phone is Android 12, and after switching to version 16.1.4, it works normally

**Comment by suryasarisa99** [comment]
> > > It works when I disable Zygisk, the problem is that I need Zygisk...
> > 
> > 
> > yes same problem, if i turn of zygisk root the app detects root. so i want to hide root and use frida at same time. do u find any solution ?
> 
> > You can try other versions of Frida. My phone is Android 12, and after switching to version 16.1.4, it works normally
> 
> > > It works when I disable Zygisk, the problem is that I need Zygisk...
> > 
> > 
> > yes same problem, if i turn of zygisk root the app detects root. so i want to hide root and use frida at same time. do u find any solution ?
> 
> You can try other versions of Frida. My phone is Android 12, and after switching to version 16.1.4, it works normally

Thanks for quick repy, it works after downgrade

**Comment by berkedel** [comment]
On my Android 15 device, I can spawn a process without encountering a timeout by following these steps.

- Set usap_pool property to false
```sh
setprop persist.sys.usap_pool_enabled false
```
- Kill the running usap processes
```sh
pkill -f usap64
```
- Restart your frida server
```sh
pkill -f frida-server && /data/local/tmp/frida-server -D
```

**Comment by JoeanAmier** [comment]
Disabling USAP is effective.

```bash
adb shell su -c "setprop persist.device_config.runtime_native.usap_pool_enabled false"
adb shell su -c "stop; start"
```
